mission_control/src/behaviors/waypoint.cpp

```diff
@@ -33,63 +33,105 @@
  *********************************************************************/
 
 // Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
-
 #include "mission_control/behaviors/waypoint.h"
 
-#include <math.h>
-#include <ros/console.h>
-#include <ros/ros.h>
-#include <stdlib.h>
-#include <string>
-#include <map>
-#include <list>
-
-#include "mission_control/Waypoint.h"
-using mission_control::Waypoint;
-using mission_control::WaypointBehavior;
-
-// WGS84 Parameters
-
-#define WGS84_A 6378137.0         // major axis
-#define WGS84_B 6356752.31424518  // minor axis
-#define WGS84_F 0.0033528107      // ellipsoid flattening
-#define WGS84_E 0.0818191908      // first eccentricity
-#define WGS84_EP 0.0820944379     // second eccentricity
-
-// UTM Parameters
-#define UTM_K0 0.9996                    // scale factor
-#define UTM_FE 500000.0                  // false easting
-#define UTM_FN_N 0.0                     // false northing, northern hemisphere
-#define UTM_FN_S 10000000.0              // false northing, southern hemisphere
-#define UTM_E2 (WGS84_E * WGS84_E)       // e^2
-#define UTM_E4 (UTM_E2 * UTM_E2)         // e^4
-#define UTM_E6 (UTM_E4 * UTM_E2)         // e^6
-#define UTM_EP2 (UTM_E2 / (1 - UTM_E2))  // e'^2
-
-WaypointBehavior::WaypointBehavior() : Behavior("waypoint", BEHAVIOR_TYPE_MSG, "/mngr/waypoint", "")
+using namespace mission_control;
+
+WaypointBehavior::WaypointBehavior(const std::string& name, const BT::NodeConfiguration& config)
+    : Behavior(name, config)
 {
-  m_altitude_ena = false;
-  m_depth_ena = false;
-  m_lat_ena = false;
-  m_long_ena = false;
-  m_wp_radius_ena = false;
-
-  m_depth = 0.0;
-  m_speed_knots = 0.0;
-  m_lat = 0;
-  m_long = 0;
-  m_wp_radius = 0;
-
-  m_depth_tol = 0.0;
-  m_duration = -1;
-
-  ros::NodeHandle nh;
-  waypoint_behavior_pub = nh.advertise<mission_control::Waypoint>("/mngr/waypoint", 1);
+  behaviorDone = false;
+  auto res = getInput<WaypointData>("value");
+  if (!res)
+  {
+    throw RuntimeError("error reading port [target]:", res.error());
+  }
+  WaypointData goal = res.value();
+
+  if (goal.depth != 0)
```

Comment:
@patapufate nit: use floating point literals when appropriate, i.e.:
```suggestion
  if (goal.depth != 0.)
```
Same below.

Reply:
Fixed